From efba9faf0c1bbf937989e03a75b9cf1d7ed59677 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 14 Dec 2022 15:28:16 +0000
Subject: [PATCH] dts: bcm2711: Add 'eee' dtparam

The 'eee' dtparam allows EEE to be disabled on the onboard Ethernet
controller. Adding 'dtparam=eee=off' to config.txt causes
'genet.eee=N' to be added to the kernel command line, which in turn
tells the Genet driver to disable EEE.

See: https://github.com/raspberrypi/linux/issues/4289

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-ds.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
@@ -2,8 +2,12 @@
 #include "bcm270x-rpi.dtsi"
 
 / {
+	chosen: chosen {
+	};
+
 	__overrides__ {
 		arm_freq;
+		eee = <&chosen>,"bootargs{on='',off='genet.eee=N'}";
 		hdmi = <&hdmi0>,"status",
 		       <&hdmi1>,"status";
 		pcie = <&pcie0>,"status";
